src/Config/HotReloadEventHandler.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.DataApiBuilder.Config
+{
+    public class HotReloadEventHandler<TEventArgs> where TEventArgs : CustomEventArgs
+    {
+        public event EventHandler<TEventArgs>? QueryManagerFactory_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? MetadataProviderFactory_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? QueryEngineFactory_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? MutationEngineFactory_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? QueryExecutor_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? MsSqlQueryExecutor_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? MySqlQueryExecutor_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? PostgreSqlQueryExecutor_ConfigChangeEventOccurred;
```

Comment:
Dont we have anything specific for CosmosDb?

Reply:
I am going to add Cosmos in its own PR because the reconfiguration could be more complicated and I don't want to delay all the other events if I run into a rabbit hole for cosmos.